display_books_aysnc-await.js

```diff
@@ -0,0 +1,69 @@
+let BookData = new Array();
+fetchBookData();
+
+async function fetchBookData() {
+  try {
+    const response = await fetch("random_books.json");
+    BookData = await response.json();
+    insertDataInHtml(BookData);
+  } catch (error) {
+    console.log("Error in Fetching data", error);
+  }
+}
+
+document.getElementById("btn-search-id").onclick = IdButton;
+document.getElementById("btn-search-genre").onclick = genreButton;
+document.getElementById("btn-search-price").onclick = priceButton;
+
+function IdButton(event) {
+  event.preventDefault();
+  const idValue = document.getElementById("input-search-id").value;
+  if (idValue.trim().length <= 0) return;
+  const matchingBooks = BookData.filter(
+    (book) => book.bookId.toString() === idValue
+  );
+  insertDataInHtml(matchingBooks);
+}
+
+function genreButton(event) {
+  event.preventDefault();
+  const genreValue = document.getElementById("input-search-genre").value;
+  if (genreValue.trim().length <= 0) return;
+  const matchingBooks = BookData.filter(
+    (book) => book.genre.toString() === genreValue
+  );
+  insertDataInHtml(matchingBooks);
+}
+
+function priceButton(event) {
+  event.preventDefault();
+  const priceValue = document.getElementById("input-search-price").value;
+  if (priceValue.trim().length <= 0) return;
+  const matchingBooks = BookData.filter(
+    (book) => book.price.toString() === priceValue
+  );
+  insertDataInHtml(matchingBooks);
+}
+
+const insertDataInHtml = (data) => {
+  data.sort((a, b) => a.price - b.price);
+  const tableBody = document.getElementById("table-all-books");
+  tableBody.innerHTML = "";
+  let row = document.createElement("tr");
+  row.innerHTML = `
+    <th>BookId</th>
+    <th>Genre</th>
+    <th>Price</th>
+    <th>Examine</th>
+   `;
+  tableBody.appendChild(row);
+  data.forEach((book) => {
+    let row = document.createElement("tr");
```

Comment:
please change the name of this variable "row" to something else

Reply:
Resolved